renku/models/datasets.py

```diff
@@ -465,3 +465,18 @@ def __attrs_post_init__(self):
 
         if not self.path:
             self.path = str(self.client.renku_datasets_path / str(self.uid))
+
+        if self.files:
+            for datasetfile in self.files:
+                if datasetfile.client is None:
+                    try:
+                        client, _, _ = self.client.resolve_in_submodules(
+                            self.client.find_previous_commit(
+                                datasetfile.path, revision='HEAD'
+                            ),
+                            datasetfile.path,
+                        )
+                    except KeyError:
```

Comment:
doesn't `KeyError` indicate that the file doesn't exist in the tree? in that case something is wrong - shouldn't we fail and let the user know in that case?

Reply:
`KeyError` is when the file wasn't found in `find_previous_commit`. There are a few unit tests that create a dataset with the `-t` parameter, which triggers the `KeyError` when the dataset is accessed unless it's handled this way. For instance `test_move_symlinks` has that problem.

Or are the unit tests broken and need to be adjusted?

Reply:
hmm... not sure - the `-t` flag allows you to specify a specific path within a git project. It's very possible that code does something funny and breaks the path in the dataset.